qiskit/pulse/frame.py

```diff
@@ -0,0 +1,77 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2021.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""Implements a Frame."""
+
+from typing import Optional, Tuple
+
+from qiskit.circuit import Parameter
+from qiskit.pulse.utils import validate_index
+from qiskit.pulse.exceptions import PulseError
+
+
+class Frame:
+    """A frame is a frequency and a phase."""
+
+    def __init__(self, identifier: str, parametric_index: Optional[Parameter] = None):
```

Comment:
Seems like this allows `Frame("Q1", Parameter("F0"))`. If I assign "0" to the parameter, does this mean "Q10"? I think this is quite complicated. This should be `identifier: str, index: Union[int, Parameter]`.

Reply:
Perhaps we should note here why `parametric_index` is necessary as a comment and create an issue to support string parameters. We could then document that `parametric_index` will eventually be deprecated.

Reply:
Probably best to make a clear distinction between the string prefix and the index. See here: ff44ff8 . If Parameters end up supporting string assignment values we can refactor to something simpler.